trunk/source/IcfgBuilder/src/de/uni_freiburg/informatik/ultimate/plugins/generator/icfgbuilder/cfg/CfgBuilder.java

```diff
@@ -1672,9 +1692,13 @@ private void mergeLocNodes(final BoogieIcfgLocation oldLocNode, final BoogieIcfg
 				mIcfg.getProcedureEntryNodes().put(mCurrentProcedureName, newLocNode);
 			}
 			if (mIcfg.getLoopLocations().remove(oldLocNode)) {
-				// if the old location was a loop location, the new one is also
+				// if the old location was a loop location, the new one is also a loop location
 				mIcfg.getLoopLocations().add(newLocNode);
 			}
+			if (mIcfg.getLocationsOfInterest().remove(oldLocNode)) {
+				// if the old location was a LOI, the new one is also a LOI
+				mIcfg.getLocationsOfInterest().add(newLocNode);
```

Comment:
I find the handling of LOIs confusing:
- Sometimes (e.g. here), `::getLocationsOfInterest` is expected to return a modifiable set. One marks a location as being of interest by calling `::add` on this set.
- Other times (see `BasicIcfg` below), the set is unmodifiable. To mark a location as being of interest, one has to pass a boolean to `addLocation`.

Could this be unified somehow?

Reply:
While building an Icfg we need more flexibility.